### PR TITLE
A strategy for replacing verbed particles with recipes

### PR DESCRIPTION
**Pull request by shans**

### CODE REVIEW

File: runtime/manifest-parser.peg [hunk 1]
```diff
@@ -858,6 +858,8 @@ TopLevelIdent
     expected(`a top level identifier (e.g. "${fixed}")`);
   }
 
+TopLevelIdentBothCases = upperIdent / lowerIdent { return text(); }
```
Comment (dstockwell) [inline runtime/manifest-parser.peg]: hmm, top level identifiers aren't meant to start with lowercase letters, maybe just inline this in Recipe as:

`name:(whiteSpace (upperIdent / lowerIdent))`

and/or a todo and we'll figure out how to clean this up (probably there aren't many recipes with names now).

File: runtime/strategies/match-recipe-by-verb.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    this._arc = arc;
+  }
+
+  async generate(strategizer) {
+    let arc = this._arc;
+    let results = Recipe.over(this.getResults(strategizer), new class extends RecipeWalker {
+      onParticle(recipe, particle) {
+        if (particle.name) {
+          // Particle already has explicit name.
+          return;
```
Comment (dstockwell) [inline runtime/strategies/match-recipe-by-verb.js]: also abort if any handle/slot connections exist

File: runtime/test/planner-tests.js [hunk 3]
```diff
@@ -698,6 +699,37 @@ describe('SearchTokensToParticles', function() {
...
+        JumpingBoots.e <- NuclearReactor.e         
+    `);
+
+    let arc = createTestArc('test-plan-arc', manifest, 'dom');
+    let strategizer = {generated: [{result: manifest.recipes[0], score: 1}]};
+    let mrv = new MatchRecipeByVerb(arc);
+    let {results} = await mrv.generate(strategizer);
+    assert.equal(results.length, 1);
+    assert.equal(results[0].result.particles.length, 0);
+    assert.deepEqual(results[0].result.toString(), 'recipe\n  JumpingBoots.e -> NuclearReactor.e\n  JumpingBoots.f -> FootFactory.f');
```
Comment (dstockwell) [inline runtime/test/planner-tests.js]: that serialization looks weird, I can't imagine we would serialize it like that. does this test really run to completion?

Comment (dstockwell) [reply runtime/test/planner-tests.js]: ohh, they are unresolved recipe constraints

File: runtime/manifest-parser.peg (hunk 1 above)
Comment (shans) [reply runtime/manifest-parser.peg]: Naming a recipe seems fine. We might eventually want both?

File: runtime/test/planner-tests.js (hunk 3 above)
Comment (shans) [reply runtime/test/planner-tests.js]: Yes. I copied the serialization that we output. You need an imagination upgrade ;)

File: runtime/manifest-parser.peg (hunk 1 above)
Comment (shans) [reply runtime/manifest-parser.peg]: Done.

File: runtime/strategies/match-recipe-by-verb.js (hunk 2 above)
Comment (shans) [reply runtime/strategies/match-recipe-by-verb.js]: done

**Pull request merged**